pyback/checkpoint.py

```diff
@@ -114,13 +122,30 @@ def iter(self):
                     stack.append((child_node, os.path.join(current_path, child_name)))
 
     @staticmethod
-    def build_checkpoint(path):
+    def build_checkpoint(path, name=None):
         root = TreeNode.build_tree_node(path, '')
-        return Checkpoint(root)
+        return Checkpoint(root, name=name)
 
     @staticmethod
     def from_json(json_str):
         tree_dict = json.loads(json_str)
 
-        return Checkpoint(TreeNode.from_dict(tree_dict['root']), datetime_from_iso_format(tree_dict['time']))
+        return Checkpoint(TreeNode.from_dict(tree_dict['root']), time=datetime_from_iso_format(tree_dict['time']), name=tree_dict['name'])
+
+
+class CheckpointMeta:
+    def __init__(self, checksum, time, name):
+        self.checksum = checksum
+        self.time = time
+        self.name = name
+
+    def to_string(self):
+        timestring = self.time.isoformat()
+        timestring += '.000000' if len(timestring) == 19 else ''
+
+        return self.checksum + '_' + timestring + '_' + self.name
+
+    @staticmethod
+    def from_string(string):
+        return CheckpointMeta(string[:16], datetime_from_iso_format(string[17:43]), string[44:])
```

Comment:
I would prefer a split on the underscores. You can specify that it should split 3 times so you do not interfere with the underscores in the name.